catalyst-gateway/bin/src/event_db/cardano/chain_state/mod.rs

```diff
@@ -85,23 +85,93 @@ impl SlotInfoQueryType {
     }
 }
 
+/// Params required for indexing follower data.
+pub(crate) struct IndexedFollowerDataParams<'a> {
+    /// Block's slot number.
+    pub slot_no: SlotNumber,
+    /// Block's epoch number.
+    pub epoch_no: EpochNumber,
+    /// Block's network.
+    pub network: &'a str,
+    /// Block's time.
+    pub block_time: DateTime,
+    /// Block's hash.
+    pub block_hash: Vec<u8>,
+}
+
+impl<'a> IndexedFollowerDataParams<'a> {
+    /// Creates a [`IndexedFollowerDataParams`] from block data.
+    pub(crate) fn from_block_data(
+        genesis_values: &GenesisValues, network: &'a str, block: &MultiEraBlock<'a>,
+    ) -> Option<Self> {
+        let epoch = match block.epoch(genesis_values).0.try_into() {
+            Ok(epoch) => epoch,
+            Err(err) => {
+                error!("Cannot parse epoch from {network:?} block {err} - skip..");
+                return None;
+            },
+        };
+
+        let wallclock = match block.wallclock(genesis_values).try_into() {
+            Ok(time) => chrono::DateTime::from_timestamp(time, 0).unwrap_or_default(),
+            Err(err) => {
+                error!("Cannot parse wall time from {network:?} block {err} - skip..");
+                return None;
+            },
+        };
+
+        let slot = match block.slot().try_into() {
+            Ok(slot) => slot,
+            Err(err) => {
+                error!("Cannot parse slot from {network:?} block {err} - skip..");
+                return None;
+            },
+        };
+
+        Some(IndexedFollowerDataParams {
+            slot_no: slot,
+            network,
+            epoch_no: epoch,
+            block_time: wallclock,
+            block_hash: block.hash().to_vec(),
+        })
+    }
+}
+
 impl EventDB {
-    /// Index follower block stream
-    pub(crate) async fn index_follower_data(
-        &self, slot_no: SlotNumber, network: Network, epoch_no: EpochNumber, block_time: DateTime,
-        block_hash: BlockHash,
+    /// Batch writes follower data.
+    pub(crate) async fn index_many_follower_data(
+        &self, values: &[IndexedFollowerDataParams<'_>],
     ) -> anyhow::Result<()> {
+        if values.is_empty() {
+            return Ok(());
+        }
+
         let conn = self.pool.get().await?;
 
-        let _rows = conn
-            .query(INSERT_SLOT_INDEX_SQL, &[
-                &slot_no,
-                &network.to_string(),
-                &epoch_no,
-                &block_time,
-                &block_hash,
-            ])
-            .await?;
+        let chunk_size = (i16::MAX / 5) as usize;
```

Comment:
Magic numbers, need to add them as a `const` with the description what do they mean.

Reply:
I'm experimenting with `COPY` statements instead of `INSERT`s to see if it speeds up the writes. Maybe we won't need this. Let me just check that before I do these changes.